### DISCUSSION TITLE
Doc Generator?

### DISCUSSION THREAD

**Discussion by amreus** [TiddlyWiki/TiddlyWiki5#6960] [General]
Does there exists a documentation/API tool which could make a list of methods defined on object in `$tw`?

Something like `yard` for Ruby, or `doxygen`?

**Comment by dubiouscript** [comment]
by the lack swift of reply's im guessing there is not one any one is aware of ...

i tried to find what the internet "thinks" 

... this old link :  https://stackoverflow.com/questions/7320464/generate-javascript-documentation-with-doxygen 
indicates its possible to use doxygen for js 

also there is https://en.wikipedia.org/wiki/JSDoc

both would (afaik) require adding some extra feilds to the comments in the js files

imho ... depending on what you are trying to find many useful tools for filtering text exist eg grep , awk  ,ect 
 which for example  make finding function definition's and function invocation's fairly easy with the correct regular expression / filter

**Comment by Jermolene** [comment]
Hi @amreus @dubiouscript we do not currently have any automated tooling for producing API documentation from the source code comments.  It would be worth doing. JSDoc probably makes sense but one could also make a case for adopting TW formatting within the comment blocks.
